core/trino-main/src/test/java/io/trino/sql/gen/BenchmarkDynamicPageFilter.java

```diff
@@ -0,0 +1,246 @@
+
+    public enum DataSet
+    {
+        INT32_RANDOM(INTEGER, (block, r) -> INTEGER.writeLong(block, r.nextInt())),
+        INT64_RANDOM(BIGINT, (block, r) -> BIGINT.writeLong(block, r.nextLong())),
+        INT64_FIXED_32K(BIGINT, (block, r) -> BIGINT.writeLong(block, r.nextLong() % 32768)), // LongBitSetFilter
+        /**/;
+
+        private final Type type;
+        private final ValueWriter valueWriter;
+
+        DataSet(Type type, ValueWriter valueWriter)
+        {
+            this.type = requireNonNull(type, "type is null");
+            this.valueWriter = requireNonNull(valueWriter, "valueWriter is null");
+        }
+
+        public TupleDomain<ColumnHandle> createFilterTupleDomain(int filterSize, boolean nullsAllowed)
+        {
+            List<Page> filterValues = createSingleColumnData(valueWriter, type, 0, filterSize);
+            ImmutableList.Builder<Object> valuesBuilder = ImmutableList.builder();
+            for (Page page : filterValues) {
+                Block block = page.getBlock(0).getLoadedBlock();
+                for (int position = 0; position < block.getPositionCount(); position++) {
+                    valuesBuilder.add(readNativeValue(type, block, position));
+                }
+            }
+            return TupleDomain.withColumnDomains(ImmutableMap.of(
+                    new TestingColumnHandle("dummy"),
+                    Domain.create(ValueSet.copyOf(type, valuesBuilder.build()), nullsAllowed)));
+        }
+
+        private List<Page> createInputTestData(
+                TupleDomain<ColumnHandle> filter,
+                double inputNullChance,
+                double nonNullsSelectivity,
+                long inputRows)
+        {
+            List<Object> nonNullValues = filter.getDomains().orElseThrow()
+                    .values().stream()
+                    .flatMap(domain -> {
+                        DiscreteSet nullableDiscreteSet = domain.getNullableDiscreteSet();
+                        return nullableDiscreteSet.getNonNullValues().stream();
+                    })
+                    .collect(toImmutableList());
+
+            // pick a random value from the filter
+            return createSingleColumnData(
+                    (block, r) -> {
+                        if (isTrue(r, nonNullsSelectivity)) {
+                            // pick a random value from the filter
+                            Object value = nonNullValues.get(r.nextInt(nonNullValues.size()));
+                            writeNativeValue(type, block, value);
+                            return;
+                        }
+                        valueWriter.write(block, r);
+                    },
+                    type,
+                    inputNullChance,
+                    inputRows);
+        }
+    }
+
+    @Setup
+    public void setup()
+    {
+        setup(MAX_ROWS);
+    }
+
+    public void setup(long inputRows)
+    {
+        // Pollute the JVM profile
```

Comment:
Do you want to pollute the trino `Type`? There are only two used here (INTEGER and BIGINT). Is it possible to use more?